src/receiver.rs

```diff
@@ -0,0 +1,41 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{to_binary, Binary, CosmosMsg, HumanAddr, StdResult, Uint128, WasmMsg};
+
+/// Snip20ReceiveMsg should be de/serialized under `Receive()` variant in a HandleMsg
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub struct Snip20ReceiveMsg {
+    pub sender: HumanAddr,
+    pub amount: Uint128,
+    pub msg: Option<Binary>,
+}
+
+impl Snip20ReceiveMsg {
+    /// serializes the message
+    pub fn into_binary(self) -> StdResult<Binary> {
+        let msg = ReceiverHandleMsg::Receive(self);
+        to_binary(&msg)
+    }
+
+    /// creates a cosmos_msg sending this struct to the named contract
+    pub fn into_cosmos_msg(
+        self, contract_code_hash: String, contract_addr: HumanAddr) -> StdResult<CosmosMsg> {
+        let msg = self.into_binary()?;
+        let execute = WasmMsg::Execute {
+            msg,
+            callback_code_hash: contract_code_hash,
+            contract_addr: contract_addr,
+            send: vec![],
+        };
```

Comment:
`callback_code_hash: contract_code_hash`  can be compacted here to `callback_code_hash` by renaming the function argument `contract_code_hash` to `callback_code_hash`.

`contract_addr: contract_addr` can be written as just `contract_addr`.